### PR TITLE
Relax rules for matching tuple names in OHI (VB)

### PR DESCRIPTION
**Pull request by jcouv**
Allows round-tripping across VB 14 and 15 with APIs involving tuples, despite VB 14 not supporting tuple element names.
For instance, you can override `Function M() As (int a, int b)` with `Function M() As ValueTuple<int, int>` or `Function M() As (int, int)`.

This is the VB change corresponding to https://github.com/dotnet/roslyn/pull/20838

### CODE REVIEW

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. This is the relaxation of OHI rules regarding tuples names, now for VB. Thanks

File: src/Compilers/VisualBasic/Portable/Symbols/SymbolExtensions.vb [hunk 1]
```diff
@@ -423,5 +423,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
         End Function
+
+        <Extension>
+        Friend Function ContainsTupleNames(member As Symbol) As Boolean
+            Select Case member.Kind
+                Case SymbolKind.Method
+                    Dim method = DirectCast(member, MethodSymbol)
+                    Return method.ReturnType.ContainsTupleNames() OrElse method.Parameters.Any(Function(p) p.Type.ContainsTupleNames())
+                Case SymbolKind.Property
+                    Return DirectCast(member, PropertySymbol).Type.ContainsTupleNames()
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Symbols/SymbolExtensions.vb]: Need to check property parameters as well. Consider extracting a method to check `ImmutableArray(Of ParameterSymbol)`.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb [hunk 2]
```diff
@@ -17745,6 +17913,26 @@ BC30402: 'evtTest3' cannot implement event 'evtTest2' on interface 'I1' because
...
+End Interface
+Class C1
+    Implements I1
+    Event evtTest3 As Action(Of (Integer, Integer)) Implements I1.evtTest1, I1.evtTest2
+End Class
+        ]]></file>
+    </compilation>, references:=s_valueTupleRefs)
+
+            CompilationUtils.AssertNoDiagnostics(compilation1)
+        End Sub
```
Comment (cston) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb]: Include tests for implementations of methods and properties as well.

File: src/Compilers/VisualBasic/Portable/Symbols/SymbolExtensions.vb [hunk 3]
```diff
@@ -429,9 +429,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
                     Dim method = DirectCast(member, MethodSymbol)
-                    Return method.ReturnType.ContainsTupleNames() OrElse method.Parameters.Any(Function(p) p.Type.ContainsTupleNames())
+                    Return method.ReturnType.ContainsTupleNames() OrElse ContainsTupleNames(method.Parameters)
                 Case SymbolKind.Property
-                    Return DirectCast(member, PropertySymbol).Type.ContainsTupleNames()
+                    Dim [property] = DirectCast(member, PropertySymbol)
+                    Return [property].Type.ContainsTupleNames() OrElse ContainsTupleNames([property].Parameters)
                 Case SymbolKind.Event
                     ' We don't check the event Type directly because materializing it requires checking the tuple names in the type (to validate interface implementations)
                     Return DirectCast(member, EventSymbol).DelegateParameters.Any(Function(dp) dp.Type.ContainsTupleNames())
```
Comment (cston) [inline src/Compilers/VisualBasic/Portable/Symbols/SymbolExtensions.vb]: `Return ContainsTupleNames(DirectCast(member, EventSymbol).DelegateParameters)`

**Comment by cston** [comment]
LGTM

**Comment by jcouv** [comment]
Thanks
@dotnet/roslyn-compiler for a second review. I'd like to merge today since I'll be OOF.

**Pull request merged**